### PR TITLE
add documentation to custom builder shell script

### LINKED ISSUE #16472: Example custom builder does not adhere to recommendations in documentation
In the [custom builder documentation](https://docs.openshift.com/container-platform/3.6/creating_images/custom.html) it is stated that

> It is recommended that you write your Custom builder image to only read the information provided by the BUILD environment variable, which conforms to an OpenShift Container Platform API version. You can then parse the required information from the build definition JSON instead of relying on the other individual environment variables provided to the builder image. 

But the [example custom builder](https://github.com/openshift/origin/blob/master/images/builder/docker/custom-docker-builder/build.sh) only uses the _not recommended_ environment variables.

Background: Getting variables from a serialized JSON structure are not trivial, so it would be tremendously valuable if the example builder gave examples on how to to that.

### PR DESCRIPTION
**Pull request by jim-minter**
fixes #16472

### CODE REVIEW

File: images/builder/docker/custom-docker-builder/build.sh [hunk 1]
```diff
@@ -1,4 +1,20 @@
...
+
+# See https://docs.openshift.org/latest/creating_images/custom.html#custom-builder-image
+# for the list of environment variables set by OpenShift before the custom
+# builder image is run.
+#
+# Although set as part of the API, the environment variables
+# SOURCE_REPOSITORY/SOURCE_URI, SOURCE_CONTEXT_DIR and SOURCE_REF can also be
+# derived from the BUILD environment variable using `jq`.  If necessary, this
+# technique can be used for extracting other values from the BUILD json from a
+# shell script.
```
Comment (bparees) [inline images/builder/docker/custom-docker-builder/build.sh]: needs a caveat that this assumes "jq" is available in your custom builder image.

File: images/builder/docker/custom-docker-builder/build.sh [hunk 2]
```diff
@@ -1,4 +1,20 @@
...
+#
+# SOURCE_REPOSITORY=$(jq -nr '(env.BUILD|fromjson).spec.source.git.uri')
+# SOURCE_URI=$(jq -nr '(env.BUILD|fromjson).spec.source.git.uri')
```
Comment (php-coder) [inline images/builder/docker/custom-docker-builder/build.sh]: `SOURCE_REPOSITORY` and `SOURCE_URI` will have the same values. Looks like, `SOURCE_REPOSITORY` should extract `spec.source.git.repository` or another member.

Comment (jim-minter) [reply images/builder/docker/custom-docker-builder/build.sh]: They are documented to be identical (API guarantee means they couldn't be changed).  https://docs.openshift.org/latest/creating_images/custom.html#custom-builder-image.

**Comment by jim-minter** [comment]
@bparees updated to add caveat.

**Pull request merged**